simpleclient_dropwizard/src/main/java/io/prometheus/client/dropwizard/DropwizardExports.java

```diff
@@ -157,6 +157,6 @@ public static String sanitizeMetricName(String dropwizardName) {
 
     @Override
     public List<MetricFamilySamples> describe() {
-      return new ArrayList<MetricFamilySamples>();
+      return collect();
```

Comment:
Why are you changing this?

Reply:
I mentioned it in #394 and here in the description above. 

If describe always returns an empty list, it causes the filtering [here](https://github.com/prometheus/client_java/blob/master/simpleclient/src/main/java/io/prometheus/client/CollectorRegistry.java#L98) to not find anything.

Reply:
I believe I disabled this as dropwizard semantics and standard usage allow for the set of metrics to be not particularly static so this feature wouldn't work well for them.

Reply:
The whole filtering feature does not work for `DropwizardExports` then. If I revert it to always return an empty list, the test I added will not work.

Reply:
You could write a small custom collector for the test.

Reply:
What would that accomplish? I am trying to fix the filtering with the actual `DropwizardExports`.

Reply:
It verifies the bugfix to the CollectorRegistry.

Filtering can't be made to reliably work with DropwizardExports, which is why it is explicitly disabled.

Reply:
If filtering can't be made to work reliably with `DropwizardExports `, then the behavior should be more meaningful than just not returning anything. When I tried the [example repo](https://github.com/RobustPerception/java_examples/blob/master/java_dropwizard/src/main/java/io/robustperception/java_examples/JavaDropwizard.java) that is using the java_client with Dropwizard and I use the filtering that the [servlet](https://github.com/prometheus/client_java/blob/master/simpleclient_servlet/src/main/java/io/prometheus/client/exporter/MetricsServlet.java#L51) obviously provides I get an empty result instead of a something telling me that filtering is not supported for Dropwizard.

Reply:
It's a limitation of how it works, we need a list of all possible metric names in advance which custom collectors cannot always provide resulting in false positives - such as you saw. It always works for direct instrumentation and usually for simpler custom collectors.

Reply:
I can understand that it is a limitation, but what I find baffling is that it does not work when using `DropwizardExports` with no clue as to why. As it is, the example simply does not work as expected with no indication why.

A simple comment explaining the situation above the `describe` method could clear it up...